### PR TITLE
[incubator/elasticsearch] Add pod disruption budgets.

### PR DESCRIPTION
**Pull request by jmcarp**
So that clusters don't become unavailable during node restarts.

cc @icereval

### CODE REVIEW

File: incubator/elasticsearch/templates/master-statefulset.yaml [hunk 1]
```diff
@@ -127,3 +127,23 @@ spec:
       resources:
         requests:
           storage: "{{ .Values.master.storage }}"
+---
```
Comment (icereval) [inline incubator/elasticsearch/templates/master-statefulset.yaml]: Please move the inline definition to a file, naming suggestion `master-pdb.yaml`

File: incubator/elasticsearch/templates/data-statefulset.yaml [hunk 2]
```diff
@@ -131,3 +131,23 @@ spec:
       resources:
         requests:
           storage: "{{ .Values.data.storage }}"
+---
```
Comment (icereval) [inline incubator/elasticsearch/templates/data-statefulset.yaml]: Inline definition to a file, naming suggestion `data-pdb.yaml`

File: incubator/elasticsearch/values.yaml [hunk 3]
```diff
@@ -38,6 +38,8 @@ master:
   storage: "4Gi"
   # storageClass: "ssd"
   antiAffinity: "soft"
+  budget:
+    minAvailable: 1
```
Comment (icereval) [inline incubator/elasticsearch/values.yaml]: Need readme documentation for this value

**Review by icereval — CHANGES_REQUESTED** [review]
A couple small requests, thanks for the help!

File: incubator/elasticsearch/templates/master-statefulset.yaml [hunk 4]
```diff
@@ -127,3 +127,23 @@ spec:
...
+{{- if .Values.master.budget.minAvailable -}}
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: "{{ template "elasticsearch.master.fullname" . }}-master-budget"
```
Comment (icereval) [inline incubator/elasticsearch/templates/master-statefulset.yaml]: Should not need the ending suffix, assuming there is no specific reason behind this choice.

File: incubator/elasticsearch/templates/data-statefulset.yaml [hunk 5]
```diff
@@ -131,3 +131,23 @@ spec:
...
+{{- if .Values.data.budget.minAvailable -}}
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: "{{ template "elasticsearch.master.fullname" . }}-data-budget"
```
Comment (icereval) [inline incubator/elasticsearch/templates/data-statefulset.yaml]: same as below, but should be `{{ template "elasticsearch.data.fullname" . }}`

**Comment by unguiculus** [comment]
Marking this as stale. Please update within one week.

**Comment by viglesiasce** [comment]
@unguiculus this is ready for review.

**Comment by unguiculus** [comment]
@jmcarp Readme update still missing.

**Comment by unguiculus** [comment]
Marking stale. Please update readme.

**Comment by unguiculus** [comment]
Closing as stale. Feel free to reopen if/when you decide to pick this up again.

**Pull request closed without merging**